src/server_manager/electron_app/index.ts

```diff
@@ -237,23 +240,69 @@ function main() {
     }
   });
 
-  // Handle request to trust the certificate from the renderer process.
-  const trustedFingerprints = new Set<string>();
-  const makeKey = (host: string, fingerprint: string) => `${host};${fingerprint}`;
-  ipcMain.on('trust-certificate', (event: IpcEvent, host: string, fingerprint: string) => {
-    trustedFingerprints.add(makeKey(host, `sha256/${fingerprint}`));
-    event.returnValue = true;
-  });
-  app.on('certificate-error', (event, webContents, url, error, certificate, callback) => {
-    event.preventDefault();
-    try {
-      const parsed = new URL(url);
-      callback(trustedFingerprints.has(makeKey(parsed.host, certificate.fingerprint)));
-    } catch (e) {
-      console.error(e);
-      callback(false);
+  // Proxy for fetch calls that require fingerprint pinning.  Assumes response is JSON.
+  ipcMain.handle(
+    'fetch-with-pin',
+    async (
+      event: Electron.IpcMainInvokeEvent,
+      url: string,
+      fingerprint: string,
+      method: string,
+      bodyJson: object,
+      bodyForm: {[id: string]: string}
+    ): Promise<object | void> => {
+      const headers: OutgoingHttpHeaders = {};
+      if (bodyJson) {
```

Comment:
This body processing logic is too high-level for this piece of code. Move it to the caller.

It would be great if this function was as close as possible to fetch, perhaps with limited parameters
We could call https://www.npmjs.com/package/node-fetch, to make that easier. But it doesn't have to be.

We seem to need in the request at least:
- URL
- method
- headers
- body (text)

And the response needs:
- status
- body text
- headers?